lib/core/client.js

```diff
@@ -929,6 +932,41 @@ function _resume (client, sync) {
   }
 }
 
+async function writeIterable (socket, contentLength, header, body) {
+  try {
+    if (contentLength === null) {
+      socket.write(`${header}transfer-encoding: chunked\r\n`, 'ascii')
+    } else {
+      socket.write(`${header}content-length: ${contentLength}\r\n\r\n`, 'ascii')
+    }
+    for await (const chunk of body) {
+      // if (typeof chunk !== 'string' && !util.sBuffer(chunk)) {
+      //   throw new Error('Body iterable must yield either string or buffer')
+      // }
+
+      const len = Buffer.byteLength(chunk)
+      if (!len) {
+        return
+      }
+
+      if (contentLength === null) {
+        socket.write(`\r\n${len.toString(16)}\r\n`, 'ascii')
+      }
+
+      if (socket.write(chunk)) {
+        continue
+      } else if (socket.destroyed) {
+        return
+      } else {
+        await EventEmitter.once(socket, 'drain')
+      }
+    }
+    socket.write('\r\n0\r\n\r\n', 'ascii')
+  } catch (err) {
+    util.destroy(socket, err)
+  }
+}
```

Comment:
Take  another look at the streams version. You need to e.g. set `client[kWriting]`.

Reply:
Missing checks with `bytesWritten`

Reply:
You basically want to almost the exact same thing but with a for await loop instead of stream events.

Reply:
okay i've gone ahead and modified writeIterable to be as close as I can make it to the stream handling. I'm failing some tests now but I'd appreciate your input on my new changes.